AppCenter/AppCenter/Internals/Context/AuthToken/MSAuthTokenContext.m

```diff
@@ -42,22 +50,49 @@ + (void)resetSharedInstance {
   sharedInstance = nil;
 }
 
-- (NSString *)authToken {
+- (nullable NSString *)authToken {
   @synchronized(self) {
     return _authToken;
   }
 }
 
-- (void)setAuthToken:(NSString *)authToken {
+- (void)setAuthToken:(NSString *)authToken withAccountId:(NSString *)accountId {
   NSArray *synchronizedDelegates;
+  BOOL isNewUser = NO;
   @synchronized(self) {
-    _authToken = authToken;
+    self.authToken = authToken;
+    isNewUser = self.homeAccountId == nil || ![self.homeAccountId isEqualToString:accountId];
```

Comment:
Is there a case that both `homeAccountId` and `accountId` are `nil`?

Reply:
@jaelim-ms `accountId` shouldn't be nil if this setter is called correctly. In all the 3 places  it get called we check for nil before calling, so it's not a problem now.

Reply:
The reason of asking this is you can simply the line like below.
`isNewUser = ![self.homeAccountId isEqualToString:accountId];`

Reply:
@jaelim-ms unless it's an Objective-C convention, I'd prefer readability over micro-optimization. 😄

Reply:
I'm saying `self.homeAccountId == nil` is redundant. It will be compared with `
[self.homeAccountId isEqualToString:accountId]`